### PR TITLE
Add Context Menu to Current Branch and Current Repository Buttons

### LINKED ISSUE #13148: Feature: Context menus on Current Repository/Branch panels
### Describe the feature or problem you’d like to solve

Feature: Add the context menus in repo/branch-selection dropdowns available on the "Current Repository" and "Current Branch" panels themselves.

### Proposed solution

This makes it easier to access context-menu options, eg copying repo/branch names, which I frequently use to paste to coworkers or into remote sessions. Currently, these panels have no context menu (right-clicking does nothing).

### PR DESCRIPTION
**Pull request by uttiya10**
Closes #13148 

## Description
- Adds Context Menu To Current Branch and Current Repository Toolbar Buttons

### Screenshots

image

## Release notes

Notes: [ADDED] Add context menu to Current Branch and Current Repository Toolbar

### CODE REVIEW

File: app/src/ui/app.tsx
Comment (niik) [inline app/src/ui/app.tsx]: Given that this method lives in the rather large `App` component I think a more suitable name for this method would be `onRepositoryToolbarButtonContextMenu`. With a generic name like this a reader would have to look for its uses to understand which context menu we're talking about.

File: app/src/ui/toolbar/branch-dropdown.tsx
Comment (niik) [inline app/src/ui/toolbar/branch-dropdown.tsx]: For consistency I'd like it if we named this `onBranchToolbarButtonContextMenu`

File: app/src/ui/app.tsx
Comment (niik) [inline app/src/ui/app.tsx]: We try to avoid relying on truthy/falsy checks, could we make this an explicit check instead and perhaps leverage optional chaining while we're at it?
```suggestion
    const repository = this.state.selectedState?.repository
    if (repository === undefined) {
      return
    }
```

File: app/src/ui/app.tsx
Comment (niik) [inline app/src/ui/app.tsx]:
```suggestion
    const externalEditorLabel = this.state.selectedExternalEditor ?? undefined
```

File: app/src/ui/branches/branch-list-item-context-menu.tsx
Comment (niik) [inline app/src/ui/branches/branch-list-item-context-menu.tsx]: I really liked the approach you took in `generateRepositoryListContextMenu` with using an interface rather than parameters. When we have multiple compatible types like this (i.e onRenameBranch and onDeleteBranch are type compatible) I'd prefer that approach.

File: app/src/ui/branches/branch-list-item-context-menu.tsx
Comment (niik) [inline app/src/ui/branches/branch-list-item-context-menu.tsx]: We typically use `new Array<T>` instead of `Array<T> = []`
```suggestion
  const items = new Array<IMenuItem>()
```

File: app/src/ui/repositories-list/repository-list-item-context-menu.ts
Comment (niik) [inline app/src/ui/repositories-list/repository-list-item-context-menu.ts]: I think it'd make sense to extract `repository` into a separate variable here to tidy things up a bit.

```suggestion
  const { repository } = config
  const missing =
    config.repository instanceof Repository && config.repository.missing
```

File: app/src/ui/repositories-list/repository-list-item-context-menu.ts
Comment (niik) [inline app/src/ui/repositories-list/repository-list-item-context-menu.ts]: We could simplify this a bunch by capturing repository into a const and only doing the `Repository` type check once

```suggestion
  const { repository } = config
  if (!(repository instanceof Repository) || !enableRepositoryAliases()) {
    return []
  }

  const verb = repository.alias == null ? 'Create' : 'Change'
  const items: Array<IMenuItem> = [
    {
      label: __DARWIN__ ? `${verb} Alias` : `${verb} alias`,
      action: () => config.onRemoveRepositoryAlias(repository),
    },
  ]

  if (repository.alias !== null) {
    items.push({
      label: __DARWIN__ ? 'Remove Alias' : 'Remove alias',
      action: () => config.onChangeRepositoryAlias(repository),
    })
  }

  return items
```

File: app/src/ui/toolbar/dropdown.tsx
Comment (niik) [inline app/src/ui/toolbar/dropdown.tsx]:
```suggestion
    this.props.onContextMenu?.(event)
```

File: app/src/ui/toolbar/button.tsx
Comment (niik) [inline app/src/ui/toolbar/button.tsx]:
```suggestion
    this.props.onContextMenu?.(event)
```

**Review by niik — CHANGES_REQUESTED** [review]
Hey @uttiya10, thanks so much for wanting to contribute! I've made some (mostly stylistic) suggestions that I'd love for you to take a look at when you've got a moment. Looking forward to having context menus on these buttons!

**Comment by uttiya10** [comment]
Thanks for the review! I pushed some changes that should hopefully resolve all your requests.

**Review by niik — APPROVED** [review]
My apologies for the long delay in getting you a second review here @uttiya10. I appreciate your patience!

I took care of merging with the latest changes from development and this is looking great now with your latest set of changes, thanks so much!

**Pull request merged**
